datagateway_api/src/search_api/helpers.py

```diff
@@ -74,8 +84,76 @@ def create_error_message(e):
     return wrapper_error_handling
 
 
+def get_score(entities, query, group, limit):
```

Comment:
I can see this function is yet to be cleaned up, so I won't review it too much. I believe the purpose of this function to to get the scoresfrom the scoring server and then add them to the entities? Would you be able to split this functionality into two functions?

Keep the request to the scoring server in this function and return the scores. Then have a second function (called `add_scores_to_entities()` or something similar), pass in `entities` and `scores`, then add the scores to the entities.

Implementing it using two functions ensures a single function has a single purpose, and doesn't try to do many things at once